commit d8cce0d5ba4a3157a7a549b9623d1ffc5820ef92
Author: Justin Stitt <justinstitt@google.com>
Date:   Wed Sep 13 20:23:02 2023 +0000

    firmware: ti_sci: refactor deprecated strncpy
    
    `strncpy` is deprecated for use on NUL-terminated destination strings [1].
    
    We should prefer more robust and less ambiguous string interfaces.
    
    A suitable replacement is `strscpy` [2] due to the fact that it guarantees
    NUL-termination on the destination buffer.
    
    It does not seem like `ver->firmware_description` requires NUL-padding
    (which is a behavior that strncpy provides) but if it does let's opt for
    `strscpy_pad()`.
    
    Link: https://www.kernel.org/doc/html/latest/process/deprecated.html#strncpy-on-nul-terminated-strings [1]
    Link: https://manpages.debian.org/testing/linux-manual-4.8/strscpy.9.en.html [2]
    Link: https://github.com/KSPP/linux/issues/90
    Cc: linux-hardening@vger.kernel.org
    Signed-off-by: Justin Stitt <justinstitt@google.com>
    Reviewed-by: Kees Cook <keescook@chromium.org>
    Link: https://lore.kernel.org/r/20230913-strncpy-drivers-firmware-ti_sci-c-v1-1-740db471110d@google.com
    Signed-off-by: Nishanth Menon <nm@ti.com>

diff --git a/drivers/firmware/ti_sci.c b/drivers/firmware/ti_sci.c
index 8444355539c6..c288ae49f91c 100644
--- a/drivers/firmware/ti_sci.c
+++ b/drivers/firmware/ti_sci.c
@@ -485,7 +485,7 @@ static int ti_sci_cmd_get_revision(struct ti_sci_info *info)
 	ver->abi_major = rev_info->abi_major;
 	ver->abi_minor = rev_info->abi_minor;
 	ver->firmware_revision = rev_info->firmware_revision;
-	strncpy(ver->firmware_description, rev_info->firmware_description,
+	strscpy(ver->firmware_description, rev_info->firmware_description,
 		sizeof(ver->firmware_description));
 
 fail: